src/matlab2tikz.m

```diff
@@ -998,7 +998,9 @@ function addCustomCode(fid, before, code, after)
     backgroundColor = get(handle, 'Color');
     if ~isNone(backgroundColor)
         [m2t, col] = getColor(m2t, handle, backgroundColor, 'patch');
-        if ~strcmp(col, 'white')
+        % Fill background if not just white or if multiple axes and current
+        % is visible
+        if ~strcmp(col, 'white') || (numel(m2t.relevantAxesHandles) > 1 && isVisible(handle))
```

Comment:
> Yes, many subplots will have white bg, as does the Matlab figure.

I support the change for this exact reason. This will change many plots on a non-whitebackground, but this is exactly what MATLAB does too.